Remove exceptions suppress

torbacz opened the pull request:
<!--- In the TITLE (↑↑↑↑ above ↑↑↑↑ **NOT HERE**) provide a general, short summary of your changes -->
<!--- Please DO NOT use references to other PR's or issues -->

## Description
<!--- Describe your changes in detail -->
<!--- Bulleted list. Full sentences. Ending with a dot. -->
-  Removed exceptions suppressions from code (and null returning when data is invalid)
- Fixed tests

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If this **fixes** OR **closes** OR  **resolves** an open issue, please link to the issue there using the template bellow (mind the pattern to link there as all issues are tracked in the Home repository) -->
<!--- **JUST** replace NNNNN with the issue number -->
- Fixes nanoFramework/Home#1114

## How Has This Been Tested?<!-- (IF APPLICABLE) -->
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots<!-- (IF APPROPRIATE): -->

## Types of changes
<!--- What types of changes does this PR introduce? Put an `x` in all the boxes that apply: -->
- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- PLEASE PLEASE PLEASE don't tick all of them just because -->
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

josesimoes reviewed (commented):
Despite lots of changes LGTM. 😅 
Unit tests are all green which is good too.

Just a nit pick on the comment.

torbacz commented on nanoFramework.Json.Test/JsonUnitTests.cs:
```diff
@@ -436,19 +436,17 @@ public void Can_serialize_deserialize_timespan_02()
+                    var _ = (JsonTestClassTimeSpan)JsonConvert.DeserializeObject(strArr[i], typeof(JsonTestClassTimeSpan));
+                    throw new Exception($"Should throw exception {nameof(InvalidCastException)}.");
+                }
+                catch (InvalidCastException) // Should throw InvalidCaseException
```
You are absolutely right!

torbacz commented:
@josesimoes 
Not sure what happed there, probable IDE just reformatted whole code in "JsonConverter" class after removing try/catch block. Haven't noticed that. In fact, there are only three changes:

- Removing try/catch block from JsonConverter.PopulateObject method
- Replaced "return null" in JsonConverter.PopulateObject method with "throw new DeserializeException"
- Adjusted TimeSpanExtensions.TryConvertFromString to throw InvalidCastException when wrong data provided (instead of returning false)

josesimoes reviewed (approved):
LGTM! Thanks.
